### ISSUE TITLE
add `consistent-import-source` rule

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod-x#11]
Allow user to define which zod import source should be used across the codebase:

### Options

- Name: `source`
- Values: 
  - `zod` (default)
  - `zod/v4`

### Configuration 

```ts
export default defineConfig({
  // ...
  rules: {
   'zod/consistent-import-source': ['error', { source: 'zod' }]
  }
  // ...
})
```

### Valid 

```ts
import * as z from 'zod';
```

### Invalid

```ts
import * as z from 'zod/v3';
```

If user apply the proposed fix the result will be:

```ts
import * as z from 'zod';
```

**Comment by marcalexiei** [comment]
https://github.com/marcalexiei/eslint-plugin-zod-x/issues/10#issuecomment-3315914086

> Maybe if we can detect that a codebase uses both v3 and v4, we could enable a rule to enforce importing from `zod/v3` and `zod/v4/core` subpaths.  

I’m not sure how safe this approach would be. 
A more flexible option might be to add an `allow` setting,
similar to [`prefer-strict-object`](https://github.com/marcalexiei/eslint-plugin-zod-x/blob/main/docs/rules/prefer-strict-object.md) to let projects configure what’s acceptable.  

---

Another option would be to flag unwanted Zod exports by leveraging the built-in `no-restricted-imports` rules (from either ESLint core or typescript-eslint):  
- https://eslint.org/docs/latest/rules/no-restricted-imports  
- https://typescript-eslint.io/rules/no-restricted-imports  

---

I’ll leave this issue open for now to gather other opinions and ideas.
